src/app/components/header/header.component.html

```diff
@@ -0,0 +1,17 @@
+<nav class="navbar navbar-dark bg-dark">
+  <a class="navbar-brand" href="/">Management App</a>
+  <ul class="nav">
+    <li class="nav-item">
+      <a class="nav-link" href="/dashboard">Dashboard</a>
```

Comment:
instead of href use routerLink="dashboard" 

into AppModule include RouterModule